ash/src/device.rs

```diff
@@ -1638,24 +1638,18 @@ impl Device {
         &self,
         pipeline_cache: vk::PipelineCache,
     ) -> VkResult<Vec<u8>> {
-        let mut data_size: usize = 0;
-        self.device_fn_1_0
-            .get_pipeline_cache_data(
+        read_into_vector(|count, data| {
+            let mut data_size = *count as usize;
+            let err_code = self.device_fn_1_0.get_pipeline_cache_data(
                 self.handle(),
                 pipeline_cache,
                 &mut data_size,
-                ptr::null_mut(),
-            )
-            .result()?;
-        let mut data: Vec<u8> = Vec::with_capacity(data_size);
-        let err_code = self.device_fn_1_0.get_pipeline_cache_data(
-            self.handle(),
-            pipeline_cache,
-            &mut data_size,
-            data.as_mut_ptr() as _,
-        );
-        data.set_len(data_size);
-        err_code.result_with_success(data)
+                data as _,
+            );
+            // TODO: Truncation might be dangerous!
+            *count = data_size as u32;
```

Comment:
I don't think we can make `read_into_vector` generic with just `Into` and `From`, perhaps this is the time to introduce https://docs.rs/num/0.4.0/num/trait.Zero.html again?

Reply:
How about `N: Default, usize: TryFrom<N>`?

Reply:
That's really neat, didn't think of that :). Switched to `N: TryInto<usize>` since it makes the trait bound a bit easier and I'd anyway write `count.try_into()` instead of `usize::try_from(count)`. Thanks!